types/jscodeshift/src/core.d.ts

```diff
@@ -1,12 +1,16 @@
-import nodes = require("ast-types/gen/nodes");
+import astTypes = require("ast-types");
+import nodePath = require("ast-types/lib/node-path");
+import types = require("ast-types/lib/types");
 import recast = require("recast");
 import Collection = require("./Collection");
 import template = require("./template");
 import VariableDeclarator = require("./collections/VariableDeclarator");
 import JSXElement = require("./collections/JSXElement");
 
 declare namespace core {
-    type Parser = recast.Parser;
+    interface Parser {
+        parse(source: string, options?: any): types.ASTNode;
+    }
```

Comment:
Not sure why this was removed as part of this change https://github.com/benjamn/recast/commit/8df3cfe6fd210e29557635877a92e305a3ad7eb9#diff-564c860307257bfac9176f554a35aa858c350a59d75c0dcd0a5d63a480805bd0

Inlining looks like a good option to me so that these types remain backwards compatible.